app/components/listing/feedback/FeedbackModal.jsx

```diff
@@ -0,0 +1,169 @@
+import React, { Fragment, useState } from 'react';
+
+import {
+  FeedbackTags, Review, SubmitMessage, tagList,
+} from './FeedbackSteps';
+
+import { images } from '../../../assets';
+import './FeedbackModal.scss';
+import { addFeedback } from '../../../utils/DataService';
+
+const getUrl = ([path, id]) => (
+  `/api/${path === 'organizations' ? 'resources' : path}/${id}/feedbacks`
+);
+
+const FeedbackModal = ({ closeModal }) => {
+  const [upvote, setUpvote] = useState(false);
+  const [downvote, setDownvote] = useState(false);
+  const [tagOptions, setTags] = useState(tagList);
+  const [review, setReview] = useState('');
+  const [step, setStep] = useState(0);
+  const [reviewRequired, setReviewRequired] = useState(false);
+  const [isSubmitted, setSubmit] = useState(false);
+
+  const handleUpvote = () => {
+    // should we reset states if user changes between upvote and downvote?
+    setStep(0);
+    setDownvote(false);
+    setReviewRequired(false);
+    setReview('');
+    setTags(tagList);
+    setUpvote(prevState => !prevState);
+  };
+
+  const handleDownvote = () => {
+    setStep(0);
+    setUpvote(false);
+    setReview('');
+    setDownvote(prevState => !prevState);
+  };
+
+  const handleSelectTag = pos => {
+    // pos = position of tag in array of tagOptions
+    const updatedTags = tagOptions.map(t => ({ ...t }));
+    updatedTags[pos].selected = !updatedTags[pos].selected;
+    setTags(updatedTags);
+    const { tag, selected } = updatedTags[pos];
+    if (tag === 'Other') setReviewRequired(selected);
+  };
+
+  const handleReview = e => {
+    e.preventDefault();
+    setReview(e.target.value);
+  };
+
+  const handleBack = () => setStep(prev => (upvote ? prev - 2 : prev - 1));
```

Comment:
Since we won't show `tagOptions` if user choose upvotes we want to skip that step.

Reply:
I think you essentially have a state machine here, and it might be clearer to express it more like a transition to states than a mathematical operation. To understand what `prev - 2` means, you also have to understand a lot about the actual sequence of steps and when it means to skip one.

A different way of expressing this would be 1) not using numbers to represent states, and 2) opting for more of a finite-state machine approach, such as the one described in this article: https://css-tricks.com/robust-react-user-interfaces-with-finite-state-machines/

We don't necessarily need to go all out like that article, but I think even simply naming the states and defining the transitions based on the state names rather than as a mathematical formula would do a lot for readability.